lib/fluent/system_config.rb

```diff
@@ -31,6 +31,33 @@ def system_config_override(opts={})
         @_system_config.send(:"#{key.to_s}=", value)
       end
     end
+
+    def listen_port_override(port=DEFAULT_LISTEN_PORT)
```

Comment:
`system_config_override` is to override instance variables for each plugin instances. Once it was overridden, following `system_config` method refers instance variable, not global system config instance.
(In other words, plugin just refer `system_config` even in tests after `system_config_override` is called in `setup` or somewhere else.)

So, there's no need to add `xxx_override` methods.